src/collapse/xdebug.rs

```diff
@@ -0,0 +1,367 @@
+use super::Collapse;
+use hashbrown::HashMap;
+use hashbrown::hash_map::RawEntryMut;
+use log::warn;
+use std::fmt;
+use std::hash::{BuildHasher, Hash, Hasher};
+use std::io::prelude::*;
+use std::io::{self, Write};
+use std::rc::Rc;
+// Ideas: str_stack and colosseum to keep allocations closer together or something.
+
+const SCALE_FACTOR: f32 = 1_000_000.0;
+static CALLS: &[&str] = &["require", "require_once", "include", "include_once"];
+
+const TRACE_START: &str = "TRACE START";
+const TRACE_END: &str = "TRACE END";
+
+/// Options for the Xdebug collapser
+#[derive(Clone, Debug, Default)]
+pub struct Options;
+
+/// A unique key for an interned string.
+#[derive(Clone, Copy, PartialEq, Eq, Hash)]
+struct Str(usize);
+
+#[derive(Clone, Copy, PartialEq, Eq, Hash)]
+enum Call {
+    WithPath(Str, Str),
+    WithoutPath(Str),
+}
+
+enum Interned<T> {
+    Old(T),
+    New(T),
+}
+
+#[derive(Default)]
+struct CallStack {
+    strings: HashMap<Rc<str>, usize>,
+    interned_string: Vec<Rc<str>>,
```

Comment:
I'm curious how much of a speedup you see from string interning? If it's significant, we might want to consider doing the same for the other stackcollapsers.

Reply:
/cc @jasonrhansen

Reply:
This change was lumped in with some other optimizations sent to me by a contributor, so it's hard to say how much of a benefit it was. Still, might be worth a try.

Reply:
I have half a mind to merge this without string interning, and then do the interning as a separate PR afterwards. It'd be a little work to separate out, but I think it would be valuable to evaluate it on its own. What do you think?